src/fb_watch.js

```diff
@@ -0,0 +1,99 @@
+let feedObserver = null;
+
+// wait for and observe FB5 feed element
+function setFeedObserver() {
+  // We are expecting to find a new feed div
+  const feed = document.querySelector(
+    'div[data-pagelet="MainFeed"]>div>div>div:not([data-adblock-monitored]):first-child'
+  );
+
+  if (feed !== null) {
+    // flag as monitored
+    feed.dataset.adblockMonitored = true;
+    const feedContainer = feed.parentNode;
+    feedObserver = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        // check if feed was reloaded without changing page
+        if (
+          mutation.target === feedContainer &&
+          mutation.addedNodes.length > 0
+        ) {
+          feedObserver.disconnect();
+          // check again for the new feed. Since the DOM has just changed, we
+          // want to wait a bit and start looking for the new div after it was
+          // rendered. We put our method at the end of the current queue stack
+          setTimeout(setFeedObserver, 0);
+        }
+        // new feed posts added
+        if (mutation.target === feed && mutation.addedNodes.length > 0) {
+          mutation.addedNodes.forEach((node) => {
+            hideIfSponsored(node);
+          });
+        }
+      });
+    });
+    // check for new feed posts
+    feedObserver.observe(feed, {
+      childList: true,
+    });
+    // check if the feed is replaced
+    feedObserver.observe(feedContainer, {
+      childList: true,
+    });
+    console.info("Monitoring feed updates", [feed]);
+  } else {
+    // no feed div was available yet in DOM. will check again
+    setTimeout(setFeedObserver, 1000);
+  }
+}
+
+/**
+ * check the element is a sponsored video
+ */
+function hideIfSponsored(e) {
+  // check sponsored video condition
+  const childNode = e.querySelector(
+    'div[aria-haspopup="menu"]:not([data-adblocked])'
+  );
+  if (childNode !== null) {
+    childNode.dataset.adblocked = true;
+    // flag a sponsored video
+    e.style.display = "none";
+    e.dataset.blocked = "sponsored";
+    console.info(`AD Blocked (div[aria-haspopup="menu"])`, [childNode, e]);
+  }
+}
+
+function onPageChange() {
+  // there's a feed div that we don't monitor yet
+  if (
+    document.querySelector(
+      'div[data-pagelet="MainFeed"]>div>div>div:not([data-adblock-monitored]):first-child'
```

Comment:
Is it because of a different in main div that this file cannot be combined with `fb5.js`?

Reply:
Well It's possible to be combined into one file `fb5.js`. The reason that I worked on a separated file was I don't want to mess the current code. I can refactor it now cuz it worked.

Reply:
Actually with this latest commit, you can remove `fb_watch.js` completely right?